entrypoint.sh

```diff
@@ -1,44 +1,115 @@
-#!/bin/sh
-set -e
-
-function checkRequired() {
-    if [ -z "${1}" ]; then
-        echo >&2 "Unable to find the ${2}. Did you set with.${2}?"
-        exit 1
-    fi
+#!/usr/bin/env bash
+set -euo pipefail
+
+bento_path=$(which bento)
+
+bento() {
+  bento_result=0
+  $bento_path --agree --email "semgrep-action@returntocorp.com" "$@" || bento_result=$?
+
+  if [[ "$1" == "check" ]]
+  then
+    echo
+    /app/semgrep-monitor "$bento_result" "--slack-url=${INPUT_SLACKWEBHOOKURL-}" || true
+    echo
+  fi
+
+  # https://github.com/returntocorp/bento/tree/cfcd3ef#exit-codes
+  # exit codes other than 0/1/2 indicate a malfunction 
+  if [[ $bento_result -ge 3 ]]
+  then
+    cat ~/.bento/last.log
+    echo
+    echo "== [ERROR] \`semgrep $*\` failed with exit code ${bento_result}"
+    echo
+    echo "This is an internal error, please file an issue at https://github.com/returntocorp/semgrep/issues/new/choose"
+    echo "and include the log output from above."
+    exit $bento_result
+  fi
+
+  return $bento_result
 }
 
-function uses() {
-    [ ! -z "${1}" ]
+handle_pull_request() {
+  # the github ref would be `refs/pull/<pr #>/merge` which isn't known by name here
+  # the github sha seems to refer to the base on re-runs
+  # so we keep our own head ref around
+  real_head_sha=$(git rev-parse HEAD)
+
+  echo
+  echo "== [1/3] going to go back to the commit you based your pull request on…"
+  echo
+  git checkout "${GITHUB_BASE_REF}"
+  git status --branch --short
+
+  echo
+  echo "== [2/3] …now adding your pull request's changes back…"
+  echo
+
+  git checkout "${real_head_sha}" -- .
+  git status --branch --short
+
+  echo
+  echo "== [3/3] …and seeing if there are any new findings!"
+  echo
+  bento init &> /dev/null
+  bento check --tool=semgrep
```

Comment:
FWIW, we could now `bento check --tool=semgrep --diff-against="${GITHUB_BASE_REF}"`

Reply:
It somehow didn't work here, reverted https://github.com/returntocorp/semgrep/runs/668320897

Reply:
```
[ERROR] 878.9339065551758 root:util - Cannot complete running command in this git state. Unknown git ref 'develop'. To run against all diffs on this repository, please use --all.
```